app/models/manageiq/providers/ansible_operation_workflow.rb

```diff
@@ -0,0 +1,109 @@
+class ManageIQ::Providers::AnsibleOperationWorkflow < Job
+  def self.create_job(env_vars, extra_vars, playbook_path, timeout: 1.hour, poll_interval: 1.minute)
+    options = {
+      :env_vars      => env_vars,
+      :extra_vars    => extra_vars,
+      :playbook_path => playbook_path,
+      :timeout       => timeout,
+      :poll_interval => poll_interval,
+    }
+
+    super(name, options)
+  end
+
+  def pre_playbook
+    # A step before running the playbook for any optional setup tasks
+    queue_signal(:run_playbook)
+  end
+
+  def run_playbook
+    env_vars, extra_vars, playbook_path = options.values_at(:env_vars, :extra_vars, :playbook_path)
+
+    response = Ansible::Runner.run_async(env_vars, extra_vars, playbook_path)
+    if response.nil?
+      queue_signal(:abort, "Failed to run ansible playbook", "error")
+    else
+      context[:ansible_runner_response] = response.dump
+
+      started_on = Time.now.utc
+      update_attributes!(:context => context, :started_on => started_on)
+      miq_task.update_attributes!(:started_on => started_on)
+
+      queue_signal(:poll_runner)
+    end
+  end
+
+  def poll_runner
+    response = Ansible::Runner::ResponseAsync.load(context[:ansible_runner_response])
+    if response.running?
+      if started_on + options[:timeout] < Time.now.utc
+        response.stop
+
+        queue_signal(:abort, "Playbook has been running longer than timeout", "error")
```

Comment:
Maybe we could log a warning here?

Reply:
Actually this is already logged by [`process_abort`](https://github.com/ManageIQ/manageiq/blob/master/app/models/job.rb#L125) and looks like this:
`ERROR -- : MIQ(ManageIQ::Providers::AnsibleOperationWorkflow#process_abort) job aborting, Playbook has been running longer than timeout`